runtime/connections.go

```diff
@@ -125,11 +125,26 @@ func (r *Runtime) Catalog(ctx context.Context, instanceID string) (drivers.Catal
 		return store, release, nil
 	}
 
-	store, ok := r.metastore.AsCatalogStore(instanceID)
+	if r.opts.CatalogConnector == "" {
```

Comment:
Should be `inst.CatalogConnector` – otherwise, metastore will always take precedence over `inst.CatalogConnector` (instance-level config should always have precedence). Removing `r.opts.CatalogConnector` would avoid some complexity that will otherwise be needed here to handle all cases.